src/main/java/org/elasticsearch/transport/netty/NettyTransport.java

```diff
@@ -110,76 +110,44 @@
-    final int connectionsPerNodePing;
-
-    final ByteSizeValue maxCumulationBufferCapacity;
-    final int maxCompositeBufferComponents;
-    */
-
-    final BigArrays bigArrays;
-    private final ThreadPool threadPool;
-    private volatile OpenChannelsHandler serverOpenChannels;
-    private volatile ClientBootstrap clientBootstrap;
+    protected static final String DEFAULT_PORT_RANGE = "9300-9400";
+    public static final String DEFAULT_PROFILE = "default";
+
+    protected final NetworkService networkService;
+    protected final Version version;
+
+    protected final boolean blockingClient;
+    protected final TimeValue connectTimeout;
+    protected final ByteSizeValue maxCumulationBufferCapacity;
+    protected final int maxCompositeBufferComponents;
+    protected final boolean compress;
+    protected final ReceiveBufferSizePredictorFactory receiveBufferSizePredictorFactory;
+    protected final int workerCount;
+    protected final ByteSizeValue receivePredictorMin;
+    protected final ByteSizeValue receivePredictorMax;
+
+    protected final int connectionsPerNodeRecovery;
+    protected final int connectionsPerNodeBulk;
+    protected final int connectionsPerNodeReg;
+    protected final int connectionsPerNodeState;
+    protected final int connectionsPerNodePing;
+
+    protected final BigArrays bigArrays;
+    protected final ThreadPool threadPool;
+    protected volatile OpenChannelsHandler serverOpenChannels;
+    protected volatile ClientBootstrap clientBootstrap;
     // node id to actual channel
-    final ConcurrentMap<DiscoveryNode, NodeChannels> connectedNodes = newConcurrentMap();
-    private final Map<String, ServerBootstrap> serverBootstraps = newConcurrentMap();
-    private final Map<String, Channel> serverChannels = newConcurrentMap();
-    private volatile TransportServiceAdapter transportServiceAdapter;
-    private volatile BoundTransportAddress boundAddress;
-    private final KeyedLock<String> connectionLock = new KeyedLock<>();
+    protected final ConcurrentMap<DiscoveryNode, NodeChannels> connectedNodes = newConcurrentMap();
+    protected final Map<String, ServerBootstrap> serverBootstraps = newConcurrentMap();
+    protected final Map<String, Channel> serverChannels = newConcurrentMap();
+    protected final Map<String, BoundTransportAddress> profileBoundAddresses = newConcurrentMap();
+    protected volatile TransportServiceAdapter transportServiceAdapter;
+    protected volatile BoundTransportAddress boundAddress;
+    protected final KeyedLock<String> connectionLock = new KeyedLock<>();
 
     // this lock is here to make sure we close this transport and disconnect all the client nodes
     // connections while no connect operations is going on... (this might help with 100% CPU when stopping the transport?)
-    private final ReadWriteLock globalLock = new ReentrantReadWriteLock();
+    protected final ReadWriteLock globalLock = new ReentrantReadWriteLock();
```

Comment:
Whoa, this lock isn't used anywhere outside of this class, why does it need to be protected?